packages/delegation-process/test/revokeDelegation.test.ts

```diff
@@ -0,0 +1,266 @@
+          when: rejectionOrRevokeDate,
+        },
+      },
+    },
+    {
+      delegationData: {
+        state: delegationState.revoked,
+        revokedAt: rejectionOrRevokeDate,
+      },
+      stamps: {
+        revocation: {
+          who: generateId<TenantId>(),
+          when: rejectionOrRevokeDate,
+        },
+      },
+    },
+  ]);
+};
+
+describe("revoke delegation", () => {
+  it("should revoke a delegation if it exists", async () => {
+    const currentExecutionTime = new Date();
+    vi.useFakeTimers();
+    vi.setSystemTime(currentExecutionTime);
+    const delegatorId = generateId<TenantId>();
+    const delegateId = generateId<TenantId>();
+    const authData = getRandomAuthData(delegatorId);
+
+    const delegationCreationDate = new Date();
+    delegationCreationDate.setMonth(currentExecutionTime.getMonth() - 2);
+
+    const delegationActivationDate = new Date();
+    delegationActivationDate.setMonth(currentExecutionTime.getMonth() - 1);
+
+    const existentDelegation = {
+      ...getMockDelegationProducer({
+        delegatorId,
+        delegateId,
+      }),
+      approvedAt: delegationActivationDate,
+      submittedAt: delegationCreationDate,
+      stamps: {
+        submission: {
+          who: delegatorId,
+          when: delegationCreationDate,
+        },
+        activation: {
+          who: delegateId,
+          when: delegationActivationDate,
+        },
+      },
+    };
+
+    await addOneDelegation(existentDelegation);
+
+    const actualDelegation = await delegationProducerService.revokeDelegation(
+      existentDelegation.id,
+      {
+        authData,
+        logger: genericLogger,
+        correlationId: randomUUID(),
+        serviceName: "DelegationServiceTest",
+      }
+    );
+
+    const expectedDelegation: Delegation = {
+      ...existentDelegation,
+      state: delegationState.revoked,
+      revokedAt: currentExecutionTime,
+      stamps: {
+        submission: {
+          who: delegatorId,
+          when: delegationCreationDate,
+        },
+        activation: {
+          who: delegateId,
+          when: delegationActivationDate,
+        },
+        revocation: {
+          who: delegatorId,
+          when: currentExecutionTime,
+        },
+      },
+    };
+
+    expect(actualDelegation).toMatchObject(expectedDelegation);
+
+    const lastDelegationEvent = await readDelegationEventByVersion(
+      actualDelegation.id,
+      1
+    );
+
+    const delegationEventPayload = decodeProtobufPayload({
+      messageType: DelegationRevokedV2,
+      payload: lastDelegationEvent.data,
+    }).delegation;
+    if (!delegationEventPayload) {
+      return fail("DelegationRevokedV2 payload not found");
+    }
+
+    const delegationFromLastEvent = fromDelegationV2(delegationEventPayload);
+
+    expect(lastDelegationEvent.version).toBe("1");
+    expect(delegationFromLastEvent).toMatchObject(expectedDelegation);
+
+    vi.useRealTimers();
```

Comment:
This can be placed in `afterAll`

Reply:
done in this [7605b94](https://github.com/pagopa/interop-be-monorepo/pull/1100/commits/7605b9460d09f6fae754a893a4818a0cb12ccfac)